src/OpenTelemetry.Api/Baggage.cs

```diff
@@ -17,6 +17,8 @@
 using OpenTelemetry.Context;
 using OpenTelemetry.Internal;
 
+#pragma warning disable RS0026 // Do not add multiple overloads with optional parameters.
```

Comment:
You should scope it down to individual methods instead of suppressing the warning for the entire file.

Consider using the [SuppressMessage](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.suppressmessageattribute?view=net-7.0) attribute for each of the methods that it warns about.